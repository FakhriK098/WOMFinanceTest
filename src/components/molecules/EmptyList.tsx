import { StyleSheet, View } from 'react-native';
import React from 'react';
import { EmptyListProps } from '../../types/components';
import { typography } from '../../theme/typography';
import Text from '../atoms/Text';
import { colors } from '../../theme/colors';

const EmptyList = ({ isDark, message }: EmptyListProps) => (
  <View style={styles.empty}>
    <Text
      size={typography.sm}
      color={isDark ? colors.text.mutedDark : colors.text.mutedLight}
    >
      {message ?? 'Belum ada data'}
    </Text>
  </View>
);

export default EmptyList;

const styles = StyleSheet.create({
  empty: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
});
