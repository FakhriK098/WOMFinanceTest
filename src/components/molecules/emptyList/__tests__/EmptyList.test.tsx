import React from 'react';
import renderer, { act } from 'react-test-renderer';

import EmptyList from '../EmptyList';

describe('EmptyList molecule', () => {
  it('matches snapshot with default message', () => {
    let tr: renderer.ReactTestRenderer;
    act(() => {
      tr = renderer.create(<EmptyList isDark={false} />);
    });
    expect(tr!.toJSON()).toMatchSnapshot();
    const text = tr!.root.find((inst) => inst.props && inst.props.children === 'Belum ada data');
    expect(text).toBeTruthy();
    act(() => tr!.unmount());
  });

  it('renders custom message', () => {
    let tr: renderer.ReactTestRenderer;
    act(() => {
      tr = renderer.create(<EmptyList isDark={true} message="Nothing here" />);
    });
    const text = tr!.root.find((inst) => inst.props && inst.props.children === 'Nothing here');
    expect(text).toBeTruthy();
    act(() => tr!.unmount());
  });
});

