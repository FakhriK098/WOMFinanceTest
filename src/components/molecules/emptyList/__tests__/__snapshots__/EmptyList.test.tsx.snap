// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyList molecule matches snapshot with default message 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#6B7280",
          "fontSize": 14,
          "fontWeight": "400",
        },
        undefined,
      ]
    }
  >
    Belum ada data
  </Text>
</View>
`;
